html/pfappserver/root/src/views/Reports/index.vue

```diff
@@ -16,72 +14,84 @@ const components = {
+import { delimiter } from './config'
 const setup = (props, context) => {
 
-  const { root: { $store } = {} } = context
+const { root: { $store } = {} } = context
 
-  const standardReports = computed(() => reportCategories().map(reportCategory => {
-    return {
-      name: i18n.t(reportCategory.name),
-      items: reportCategory.reports.map(report => {
-        return {
-          name: i18n.t(report.name),
-          path: `/reports/standard/chart/${report.tabs[0].path}`,
-          icon: report.chart ? 'chart-pie' : null
-        }
-      })
-    }
-  }))
+  const reports = ref([])
+  onMounted(() => $store.dispatch('$_reports/all')
+    .then(_reports => {
+      reports.value = _reports
+    })
+  )
 
-  const otherReports = ref([])
-  const dynamicReports = ref([])
-  onMounted(() => {
-    $store.dispatch('$_reports/all').then(reports => {
-      const sorted = [...(new Set(reports))] // dereferenced (prevents `sort` mutation)
-        .sort((a, b) => i18n.t(a.description).localeCompare(i18n.t(b.description)))
+  const _depth = (item, depth = 0) => {
+    let d = depth
+      const { children = {} } = item
+      Object.keys(children).forEach(key => {
+          depth = Math.max(depth, _depth(children[key], d + 1))
+      })
+    return depth
+  }
 
-      otherReports.value = sorted.filter(report => { return report.type === 'builtin' })
-        .map(report => {
+  const _struct = (associated, parents = 0) => {
+    return Object.keys(associated).map(key => {
+      const { children = {}, id, charts = '', date_field, has_date_range, searches } = associated[key]
+      const icons = []
+      if (searches)
+        icons.push('search')
+      if (date_field || has_date_range)
+        icons.push('calendar-alt')
+      if (charts)
+        icons.push('chart-pie')
+      const depth = _depth(associated[key])
+      switch (true) {
+        case depth === 2 && parents === 0:
+        case depth === 1 && parents === 0:
           return {
-            name: i18n.t(report.description),
-            path: `/reports/dynamic/chart/${report.id}`,
-            saveSearchNamespace: `dymamicReports::${report.id}`
+            name: i18n.t(key), // i18n defer
+            collapsable: true,
+            items: _struct(children, parents + 1) // recursive
           }
-        })
-
-      dynamicReports.value = sorted.filter(report => { return !report.type || report.type !== 'builtin' })
-        .map(report => {
+          // break
+        case depth === 1 && parents === 1:
           return {
-            name: i18n.t(report.description),
-            path: `/reports/dynamic/chart/${report.id}`,
-            saveSearchNamespace: `dymamicReports::${report.id}`
+            name: i18n.t(key), // i18n defer
+            items: _struct(children, parents + 1) // recursive
           }
-        })
-      })
-  })
+          // break
+        case depth === 0:
+          return {
+            name: i18n.t(key), // i18n defer
```

Comment:
The `key` should not be translated at this step -- it should translated and returned **by the server**. Remove the call to `i18n` and the commented instruction.